android-core/src/main/java/com/mparticle/networking/NetworkOptions.java

```diff
@@ -144,10 +147,13 @@ public Builder addDomainMapping(@Nullable DomainMapping domain) {
                 domainMappings = new HashMap<Endpoint, DomainMapping>();
             }
             if (domainMappings.containsKey(domain.getType())) {
-                try {
-                    Logger.warning("Duplicate DomainMapping submitted, DomainMapping:\n" + domain.toJson().toString(4) + "\n will overwrite DomainMapping:\n" + domain.toJson().toString(4));
-                } catch (JSONException e) {
-                    e.printStackTrace();
+                if (!loggedDomainTypes.contains(domain.getType())) {
+                    try {
+                        Logger.warning("Duplicate DomainMapping submitted, DomainMapping:\n" + domain.toJson().toString(4) + "\n will overwrite DomainMapping:\n" + domain.toJson().toString(4));
+                        loggedDomainTypes.add(domain.getType());
+                    } catch (JSONException e) {
+                        e.printStackTrace();
```

Comment:
Not sure if this is ideal, shouldn't this be 
```suggestion
                        Logger.error(e);
```